### DISCUSSION TITLE
Is there a way to set the environment variable from code and have Prisma read that variable?

### DISCUSSION THREAD

**Discussion by walterstamm** [prisma/prisma#16012] [Q&A] [answered]
I am working with AWS secrets, and from there I want to take the connection data, I try to build the environment variable for the connection to the database but Prisma takes as if the variable does not exist

![image](https://user-images.githubusercontent.com/68483433/198068823-d74a6e06-f7bc-4e3b-b29a-295ebca0f14e.png)

And this is the output 

![image](https://user-images.githubusercontent.com/68483433/198069616-eaff00c8-bbd7-4371-bb42-53b332e2010f.png)

**Comment by austincrim** [comment, marked as answer]
Hey @walterstamm 👋,

You can pass the connection string at runtime using the `PrismaClient` constructor.

```ts
const prisma = new PrismaClient({ datasources: { db: { url: 'my connection string' } } })
```

**Reply by janpio** [reply]
The error message indicates that @walterstamm is already trying that, but his screenshot does not show where and how he actually tries to instantiate Prisma. The terminal output also does not show the `console.log` of the first screenshot actually working - so not sure if that even is active before Prisma tries to start.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
